### PR TITLE
Form validation: fix french alphanumeric validation and french error message

### PR DESCRIPTION
**Pull request by polmih**
Form validation: fix french alphanumeric validation and french error message ( the error message in french was saying it is accepting the spaces too and its not true).

Related to WET-259

### CODE REVIEW

**Comment by RobJohnston** [comment]
This matches the changes to the jQuery Validation plugin done here 10 years ago:
https://github.com/jquery-validation/jquery-validation/pull/375/files

However, their French translation was never updated:
https://github.com/jquery-validation/jquery-validation/blob/master/src/localization/messages_fr.js

**Comment by polmih** [comment]
> This matches the changes to the jQuery Validation plugin done here 10 years ago: https://github.com/jquery-validation/jquery-validation/pull/375/files
> 
> However, their French translation was never updated: https://github.com/jquery-validation/jquery-validation/blob/master/src/localization/messages_fr.js

Although their regex will not allow french characters :)

**Comment by duboisp** [comment]
Pre-approved upon successful code review and testing.

@polmih can you submit the same fix upstream in the jQuery validation plugin. and add reference in your code here. Thank you

**Comment by polmih** [comment]
> 

PR sent : https://github.com/jquery-validation/jquery-validation/pull/2456

**Comment by RobJohnston** [comment]
Changing the regex to include French characters surely isn't going to fly with jQuery Validation.  When I earlier said that they made the same change, I was just referencing the translation of the error message.

Even WET-BOEW supports more than 2 languages.  Not sure that the regex should be changed. 🤔

**Comment by duboisp** [comment]
Pre-approved upon successful code review and testing.

Thanks @polmih for the PR

**Review by duboisp — APPROVED** [review]
Tested locally and work as expected

**Comment by duboisp** [comment]
@polmih after considering @RobJohnston comment, let's localize that regular expression override for only the French pages.

@RobJohnston I do agree that might don't fly with jQuery Validation group, but it can help other.

File: src/plugins/formvalid/formvalid.js [hunk 1]
```diff
@@ -93,6 +93,19 @@ var componentName = "wb-frmvld",
...
 						i, len, validator;
 
+
+					// alphanumeric regex is changed to allow french characters;
+					$.validator.addMethod( "alphanumeric", function( value, element ) {
+						return this.optional( element ) || /^[a-zàâçéèêëîïôûùüÿæœ0-9_]+$/i.test( value );
+					}, "Letters, numbers, and underscores only please." );
+
+					// // error french text is adjusted to remove the word "spaces"
+					if ( wb.lang === "fr" ) {
```
Comment (duboisp) [inline src/plugins/formvalid/formvalid.js:103]:
```suggestion

					if ( wb.lang === "fr" ) {
						// alphanumeric regex is changed to allow french characters;
						$.validator.addMethod( "alphanumeric", function( value, element ) {
							return this.optional( element ) || /^[a-zàâçéèêëîïôûùüÿæœ0-9_]+$/i.test( value );
						}, "Letters, numbers, and underscores only please." );

					// error french text is adjusted to remove the word "spaces"
```

**Review by duboisp — CHANGES_REQUESTED** [review]

File: src/plugins/formvalid/formvalid.js (hunk 1 above)
Comment (polmih) [reply src/plugins/formvalid/formvalid.js]: Done. Thanks

**Comment by duboisp** [comment]
Pre-approved upon successful local testing. Code review is ok.

**Review by duboisp — APPROVED** [review]
Tested locally and reviewed the code. All good

**Pull request merged**